### PR TITLE
Null safety migration of packages/flutter_tools/bin

### PR DESCRIPTION
**Pull request by liamappelbe**
Null safety migration of packages/flutter_tools/bin

### CODE REVIEW

**Comment by liamappelbe** [comment]
Blocked on [dwds migration](https://github.com/dart-lang/webdev/issues/1327)

**Comment by liamappelbe** [comment]
Tests are passing

File: packages/flutter_tools/bin/fuchsia_tester.dart [hunk 1]
```diff
@@ -162,7 +160,7 @@ Future<void> run(List<String> args) async {
       // package (i.e. contains lib/ and test/ sub-dirs). In some cases,
       // test files may appear to be in the root directory.
       if (coverageDirectory == null) {
-        globals.fs.currentDirectory = testDirectory.parent;
+        globals.fs.currentDirectory = testDirectory?.parent;
```
Comment (christopherfujino) [inline packages/flutter_tools/bin/fuchsia_tester.dart:163]: does it make sense to set `fs.currentDirectory` to a null?

File: packages/flutter_tools/bin/fuchsia_tester.dart [hunk 2]
```diff
@@ -162,7 +160,7 @@ Future<void> run(List<String> args) async {
       // package (i.e. contains lib/ and test/ sub-dirs). In some cases,
       // test files may appear to be in the root directory.
       if (coverageDirectory == null) {
-        globals.fs.currentDirectory = testDirectory.parent;
+        globals.fs.currentDirectory = testDirectory!.parent;
       } else {
         globals.fs.currentDirectory = testDirectory;
```
Comment (christopherfujino) [inline packages/flutter_tools/bin/fuchsia_tester.dart:165]:
```suggestion
        globals.fs.currentDirectory = testDirectory!;
```

Comment (liamappelbe) [reply packages/flutter_tools/bin/fuchsia_tester.dart:165]: I tried that, but the analyzer complains that this is an unnecessary assertion.

Comment (christopherfujino) [reply packages/flutter_tools/bin/fuchsia_tester.dart:165]: I see, I guess because `set currentDirectory` takes a dynamic. This is fine.

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**